### PR TITLE
Feature/lps 88329

### PR DESCRIPTION
**Pull request by andresfulla**
Hey Kevin! Can you take a look?

 I'll wait to have https://github.com/andresfulla/liferay-portal/tree/LPS-88575 in master to PR this one. But I think it can be useful if you can take a look. Any webpack thing should be coming from `LPS-88575`

### CODE REVIEW

File: modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/ContributorsBuilder.es.js [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+	}
+);
+const propertyTypes = PropTypes.shape(
+	{
+		boolean: PropTypes.arrayOf(PropTypes.string).isRequired,
+		date: PropTypes.arrayOf(PropTypes.string).isRequired,
+		number: PropTypes.arrayOf(PropTypes.string).isRequired,
+		string: PropTypes.arrayOf(PropTypes.string).isRequired
+	}
+);
```
Comment (thektan) [inline modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/ContributorsBuilder.es.js]: Let's add new lines here. Constants usually get new lines in between them too, unless the constants are related like

```javascript
// no new line
const labelOne = 'one';
const labelTwo = 'two';
```

File: modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/CriteriaBuilder.es.js [hunk 2]
```diff
@@ -20,55 +16,55 @@ const CRITERION_SHAPE = {
...
-		).isRequired,
-		supportedPropertyTypes: PropTypes.object
-	};
+		}
+	),
+	editing: PropTypes.bool.isRequired,
+	id: PropTypes.number.isRequired,
+	modelLabel: PropTypes.string,
+	onChange: PropTypes.func,
+	onEditionToggle: PropTypes.func,
```
Comment (thektan) [inline modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/CriteriaBuilder.es.js]: Can we rename this prop to `onEditToggle`?

File: modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/ContributorsBuilder.es.js [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+							}
+						/>
+					}
+				</div>
+			</div>
+		);
+	}
+}
+
+const conjuctions = PropTypes.shape(
```
Comment (thektan) [inline modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/ContributorsBuilder.es.js]: Typo `conjunctions` (missing `n`)

File: modules/apps/segments/segments-web/test/js/components/shared/ClayToggle.es.js [hunk 4]
```diff
@@ -6,12 +6,14 @@ describe(
...
 		afterEach(cleanup);
-
+		const mockOnChange = () => {};
 		it(
 			'should render',
 			() => {
 				const {asFragment} = render(
-					<ClayToggle />
+					<ClayToggle
+						onChange={mockOnChange}
```
Comment (thektan) [inline modules/apps/segments/segments-web/test/js/components/shared/ClayToggle.es.js]: We can mock empty functions with `jest.fn()`.

```
onChange={jest.fn()}
```

File: modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/ContributorsBuilder.es.js [hunk 5]
```diff
@@ -0,0 +1,286 @@
...
+									[selectedProperty && selectedProperty.name]
+								)
+							}
+						/>
+					}
+				</div>
+			</div>
+		);
+	}
+}
```
[the remark this answers is not in the document]
Comment (andresfulla) [reply modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/ContributorsBuilder.es.js]: Thanks! I think I got the idea and fixed all of those.

**Comment by andresfulla** [comment]
@thektan Can you review this and CC Bryce (https://github.com/bryceosterhaus/liferay-portal/pull/13#issuecomment-453220143) if all it's good? Thanks

**Comment by thektan** [comment]
Made some changes and sent @andresfulla a new pr: https://github.com/andresfulla/liferay-portal/pull/11

**Pull request closed without merging**